### PR TITLE
Fix builtin security check.

### PR DESCRIPTION
**Pull request by alonh5**
# Fix builtin security check

## Description

Fix builtin security check to match python VM.

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: src/vm/runners/builtin_runner/mod.rs [hunk 1]
```diff
@@ -303,17 +306,14 @@ impl BuiltinRunner {
...
-                let offset = offset + j;
-                match offsets_iter.next_if_eq(&offset) {
-                    Some(_) => {}
-                    None => {
+                let offset = offset_base + j;
+                match offsets.contains(&offset) {
+                    true => {}
+                    false => {
                         missing_offsets.push(offset);
                     }
```
Comment (Oppen) [inline src/vm/runners/builtin_runner/mod.rs:318]:
```suggestion
                if !offsets.contains(&offset) {
                    missing_offsets.push(offset);
```

File: src/vm/runners/builtin_runner/mod.rs [hunk 2]
```diff
@@ -303,17 +306,14 @@ impl BuiltinRunner {
...
         for i in 0..n as usize {
-            let offset = cells_per_instance as usize * i;
+            let offset_base = cells_per_instance as usize * i;
             for j in 0..n_input_cells as usize {
-                let offset = offset + j;
-                match offsets_iter.next_if_eq(&offset) {
-                    Some(_) => {}
-                    None => {
+                let offset = offset_base + j;
+                match offsets.contains(&offset) {
```
Comment (Oppen) [inline src/vm/runners/builtin_runner/mod.rs:314]: This makes the complexity go from `O(n_input_cells)` to `O(n_input_cells*offsets.len()`. Could this be called often?

Comment (alonh5) [reply src/vm/runners/builtin_runner/mod.rs]: I changed it back to `O(n*n_input_cells)`.

**Review by Juan-M-V — APPROVED** [review]

File: src/vm/runners/builtin_runner/mod.rs [hunk 3]
```diff
@@ -289,7 +289,10 @@ impl BuiltinRunner {
             })
             .collect::<Vec<_>>();
 
-        let n = div_floor(offsets.len(), cells_per_instance as usize);
+        let n = match offsets.iter().max() {
+            Some(x) => div_floor(*x, cells_per_instance as usize) + 1,
+            None => 0,
+        };
```
Comment (Juan-M-V) [inline src/vm/runners/builtin_runner/mod.rs:295]: A `map_or` could shorten this match to `offsets.iter().max().map_or(0, |v| div_floor(*x, cells_per_instance as usize) + 1)`

**Review by Oppen — APPROVED** [review]

**Pull request merged**